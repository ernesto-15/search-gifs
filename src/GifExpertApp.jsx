import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = ['Rick and Morty'] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h1>Gif Expert App</h1>
      <AddCategory setCategories={setCategories} />
      {categories.map((category, key) => (
        <GifGrid key={category} category={category} />
        // <li key={key}>{category}</li>
      ))}
    </>
  );
};

export default GifExpertApp;
