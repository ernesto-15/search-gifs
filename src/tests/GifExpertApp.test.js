import React from 'react';
import GifExpertApp from '../GifExpertApp.jsx';
import { shallow } from 'enzyme';

describe('tests in GifExpertApp component', () => {
  test('should show component', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show categories', () => {
    const categories = ['Messi', 'Ronaldo'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
