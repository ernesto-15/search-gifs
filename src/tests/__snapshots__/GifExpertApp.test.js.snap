// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifExpertApp component should show categories 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="Messi"
    key="Messi"
  />
  <GifGrid
    category="Ronaldo"
    key="Ronaldo"
  />
</Fragment>
`;

exports[`tests in GifExpertApp component should show component 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="Rick and Morty"
    key="Rick and Morty"
  />
</Fragment>
`;
